src/render/draw_raster.js

```diff
@@ -28,6 +28,7 @@ function drawRaster(painter: Painter, sourceCache: SourceCache, layer: RasterSty
     const stencilMode = StencilMode.disabled;
     const colorMode = painter.colorModeForRenderPass();
     const minTileZ = coords.length && coords[0].overscaledZ;
+    const align = !painter.options.moving && (Math.round(painter.transform.zoom) === painter.transform.zoom);
```

Comment:
I think the ` && (Math.round(painter.transform.zoom) === painter.transform.zoom)` part isn't necessary. Choosing the aligning matrix isn't affecting nearest neighbor vs. bilinear filtering, so it's safe to use the aligned matrix even for fractional zoom levels (in fact, this produces better bilinear filtering, as indicated by the old test case image of the render test updated in this PR).

Another behavior that I'm observing at least in Firefox is that `moving` isn't reset to false correctly when the mouse is released. Another possibility is that it _is_ reset correctly, but no render is triggered when its state changes, leaving the map in a blurry state when it is rotated.

Reply:
(Noting that this wasn't a problem before because we always used aligned rendering)